src/Native/System.Native/pal_networkchange.cpp

```diff
@@ -52,7 +52,8 @@ extern "C" NetworkChangeKind SystemNative_ReadSingleEvent(int32_t sock)
     iovec iov = {buffer, sizeof(buffer)};
     sockaddr_nl sanl;
     msghdr msg = {reinterpret_cast<void*>(&sanl), sizeof(sockaddr_nl), &iov, 1, NULL, 0, 0};
-    ssize_t len = recvmsg(sock, &msg, 0);
+    ssize_t len;
+    while (CheckInterrupted(len = recvmsg(sock, &msg, 0)));
```

Comment:
I dislike assignment-as-expression because it's easy to get wrong, is easily visually confused with comparison, and hides an important side-effecting operation inside of a larger expression. Overall I find it less readable than the alternatives. Even

`do { len = recvmsg(sock, &msg, 0); } while (CheckInterrupted(len));`

or

`for (len = recvmsg(sock, &msg, 0); CheckInterrupted(len); len = recvmsg(sock, &msg, 0));`

would be preferable IMO.